waf_dl.go

```diff
@@ -99,11 +99,15 @@ func (waf *wafDl) wafGetVersion() string {
 	return gostring(cast[byte](waf.syscall(waf.getVersion)))
 }
 
-func (waf *wafDl) wafInit(ruleset *wafObject, config *wafConfig, info *wafObject) wafHandle {
+// wafInit initializes a new WAF with the provided ruleset, configuration and info objects. A
+// cgoRefPool ensures that the provided input values are not moved or garbage collected by the Go
+// runtime during the WAF call.
+func (waf *wafDl) wafInit(ruleset *wafObject, config *wafConfig, info *wafObject, refs cgoRefPool) wafHandle {
```

Comment:
If you do it here you can also do it in `wafRun` I guess

Reply:
Except wafRun requires some of the data to live as long as the context does, but not all of them... I actually have some (perhaps better?) ideas there, so I'll probably end up making another PR with changes to the `cgoRefPool` ergonomics...